src/State/Provider/BackedEnumProvider.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace ApiPlatform\State\Provider;
+
+use ApiPlatform\Metadata\CollectionOperationInterface;
+use ApiPlatform\Metadata\Operation;
+use ApiPlatform\State\ProviderInterface;
+
+final class BackedEnumProvider implements ProviderInterface
+{
+    public function __construct(private readonly ProviderInterface $decorated)
+    {
+    }
+
+    public function provide(Operation $operation, array $uriVariables = [], array $context = []): object|array|null
+    {
+        $className = $operation->getClass();
+        if (\is_string($className) && enum_exists($className)) {
+            if ($operation instanceof CollectionOperationInterface) {
+                return $className::cases();
+            }
+
+            $id = $uriVariables['id'] ?? $uriVariables['value'] ?? null;
```

Comment:
probably that we can always use `id` here

Reply:
If an enum does not specify an identifier, then the `LinkFactory` marks `value` as the identifier field.

Reply:
e.g. the `Status` enum example above:
![image](https://github.com/api-platform/core/assets/1427081/24e1b686-0e41-4adc-a749-4c67e627115a)

Reply:
We can use `id` as uri template and use the `value` in the `identifiers` mapping if needed